app/containers/ActionSheet/ActionSheetContentWithInputAndSubmit/index.tsx

```diff
@@ -114,25 +132,44 @@ const ActionSheetContentWithInputAndSubmit = ({
 					placeholder={placeholder}
 					onChangeText={value => setInputValue(value)}
 					onSubmitEditing={() => {
-						// fix android animation
-						setTimeout(() => {
-							hideActionSheet();
-						}, 100);
-						if (inputValue) onSubmit(inputValue);
+						if (canSubmit) {
```

Comment:
This logic is wrong, because if I haven't typed anything, it can submit to disappear the action sheet.

Reply:
Actually, when we press the "submit" from the keyboard, it's expected that it really submit the data. So, I added a prop "isObligatory" to conditional.